### ISSUE TITLE
Fix S2094 FP: Allow empty records

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#9583]
### Description

I use a combination of private ctors and nested records to simulate discriminated unions. However, I receive a warning about this.

### Repro steps

```c#
internal abstract record Origin
{
    private Origin()
    {
    }

    public sealed record Client : Origin; // S2094 Remove this empty record, write its code or make it an "interface".

    public sealed record Manual(UserId OriginatorId) : Origin;
}
```

### Expected behavior

No warnings are reported.

### Actual behavior

Warning S2094 is reported.

### Related information

* C#/VB.NET Plugins version: 9.21.0.86780

**Comment by sebastien-marichal** [comment]
Hello @hankovich,

This seems very similar to #7709.
Which has been fixed in 9.23.

I was not able to reproduce the issue with the latest version 9.30.

Did you try using a more recent version of our analyzer?

**Comment by sebastien-marichal** [comment]
I was able to reproduce with 9.22.0.87781 and the issue disappears in 9.23.0.88079.
I am closing this issue as it has already been fixed.
